modules/social_features/social_follow_content/social_follow_content.install

```diff
@@ -5,67 +5,34 @@
  * Install, update and uninstall functions for the social_follow_content module.
  */
 
-use Drupal\user\Entity\Role;
-
 /**
  * Implements hook_install().
  *
  * Perform actions related to the installation of social_follow_content.
  */
 function social_follow_content_install() {
-
-  // Set some default permissions.
-  _social_follow_content_set_permissions();
-}
-
-/**
- * Function to set permissions.
- */
-function _social_follow_content_set_permissions() {
-  $roles = Role::loadMultiple();
-
-  /** @var \Drupal\user\Entity\Role $role */
-  foreach ($roles as $role) {
-    if ($role->id() === 'administrator') {
-      continue;
-    }
-
-    $permissions = _social_follow_content_get_permissions($role->id());
-    if (!empty($permissions)) {
-      user_role_grant_permissions($role->id(), $permissions);
-    }
-  }
-}
-
-/**
- * Build permissions.
- *
- * @param string $role
- *   The role.
- *
- * @return array
- *   Returns an array containing the permissions.
- */
-function _social_follow_content_get_permissions($role) {
-  // Anonymous.
-  $permissions['anonymous'] = [];
-
-  // Authenticated.
-  $permissions['authenticated'] = array_merge($permissions['anonymous'], [
-    'flag follow_content',
-    'unflag follow_content',
-  ]);
-
-  // Content manager.
-  $permissions['contentmanager'] = array_merge($permissions['authenticated'], []);
-
-  // Site manager.
-  $permissions['sitemanager'] = array_merge($permissions['contentmanager'], []);
-
-  if (isset($permissions[$role])) {
-    return $permissions[$role];
-  }
-  return [];
+  // Grant the default permissions for this feature.
+  user_role_grant_permissions(
+    'verified',
+    [
+      'flag follow_content',
```

Comment:
Depending on the outcome of the discussion if authenticated user can view only "public" vs also "community" content, this may also need to change.

Reply:
From PR description we can get that unverified user can't actively engage with the entire community yet. So, I think it also means that unverified users shouldn't have access to community content.